src/core/geometries/collision.js

```diff
@@ -0,0 +1,50 @@
+import { halfPlaneTest, pointInRectangle, pointInTriangle, checkSign } from '../../utils/geometry';
+
+export function triangleCollides (triangle, viewport, viewportAABB) {
+    if (_isFeatureGreaterThanViewport(triangle, viewport)) {
+        return true;
+    }
+
+    if (_isTriangleInViewport(triangle, viewportAABB)) {
```

Comment:
This seems to me a more frequent and more efficient check, so I'd placed it in first place (very minor optimization, but...)

Reply:
True! ✨